linux/storage/loopback.sls

```diff
@@ -0,0 +1,45 @@
+{%- from "linux/map.jinja" import storage with context %}
+
+{%- if storage.get('enabled', False) %}
+
+{%- for device, loopback in storage.loopback.iteritems()|dictsort %}
+
+{%- if loopback.get('enabled', True) %}
+
+{{ salt['file.dirname'](loopback.local_file) }}:
+  file.directory:
+  - makedirs: true
+  - require_in:
+    - file: {{ loopback.local_file }}
+
+{{ loopback.local_file }}:
+  cmd.run:
+  - name: "truncate --size {{ loopback.size|default('1G') }} {{ loopback.local_file }}"
+  - creates: {{ loopback.local_file }}
+
+{%- if grains.get('init', None) == 'upstart' %}
+loopback_{ device }}_init_script:
+  file.managed:
+{%- if grains.get('init', None) == 'upstart' %}
+  - name: /etc/init/setup-loopback-{{ device }}.conf
+  - source: salt://linux/files/setup-loopback-device.upstart
+{%- else %}
+  - name: /etc/systemd/system/setup-loopback-{{ device }}.service
+  - source: salt://linux/files/setup-loopback-device.systemd
+{%- endif %}
+  - template: jinja
+  - defaults:
+    local_file: {{ loopback.local_file }}
+    device_name: "/dev/loop{{ loop.index0 }}"
+
+loopback_{ device }}_start:
+  service.start:
+  - unless: "losetup -a | egrep 'loop{{ loop.index0 }}$'"
```

Comment:
You should maybe set `enabled: true` here to start on boot. And don't know if that unless condition is needed, init script should be idempotent in this case and shouldn't do anything if loopback device already exists.